src/combine-results.ts

```diff
@@ -94,34 +93,71 @@ async function run(): Promise<void> {
       title,
     });
 
-    const csvs: string[] = [];
-    const resultsMd = await Promise.all(
+    const csvs: string[] = await Promise.all(
       resultArtifacts.map(async function (response) {
         const csv = path.join(response.downloadPath, "results.csv");
         const csvDest = path.join("results", response.artifactName);
         await mv(csv, csvDest);
-        csvs.push(csvDest);
-
+        return csvDest;
+      })
+    );
+    const resultsMd = await Promise.all(
+      resultArtifacts.map(async function (response) {
         const repoName = fs.readFileSync(
           path.join(response.downloadPath, "nwo.txt"),
           "utf-8"
         );
-        const output = await getExecOutput("wc", ["-l", csvDest]); // TODO: preferably we would do this during results interpretation
-        const results = parseInt(output.stdout.trim(), 10) - 1;
+        const resultCount = parseInt(
+          fs.readFileSync(
+            path.join(response.downloadPath, "resultcount.txt"),
+            "utf-8"
+          ),
+          10
+        );
 
-        if (results > 0) {
+        if (resultCount > 0) {
           const md = path.join(response.downloadPath, "results.md");
           const comment = await octokit.rest.issues.createComment({
             owner: context.repo.owner,
             repo: context.repo.repo,
             issue_number: issue.data.number,
             body: fs.readFileSync(md, "utf8"),
           });
-          return `| ${repoName} | [${results} result(s)](${comment.data.html_url}) |`;
+          return `| ${repoName} | [${resultCount} result(s)](${comment.data.html_url}) |`;
         }
         return `| ${repoName} | _No results_ |`;
       })
     );
+    const resultsIndex = await Promise.all(
+      resultArtifacts.map(async function (response) {
```

Comment:
Again, not critical, but a custom type to build and then serialize would be useful internal documentation.

I also think there is an interesting test opportunity here where you factor out this part into another file (you can't test against the "main" action file because it tries to run it!) and then call `runQuery` followed by this logic pointed out the output directory.